x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -360,3 +366,17 @@ func createMetricDataQuery(id string, metricName string, periodInSec int, dimens
 	}
 	return
 }
+
+func checkInstanceStatus(instanceState ec2.InstanceStateName, launchTime *time.Time, instanceID string) bool {
+	if instanceState != ec2.InstanceStateNameRunning {
+		logp.NewLogger(aws.ModuleName).Infof("Instance %s is not in running state. Skip data collection for this EC2 instance.", instanceID)
+		return false
+	}
+	currentTime := time.Now()
+	diff := currentTime.Sub(*launchTime)
+	if diff.Minutes() < 10 {
```

Comment:
Alternative suggestion: We start to report some data as soon as the instance is there as I think that is already valuable for inventory. The cloud metadata is already there. We can add a field that indicates this status.

For each request, we check if the metrics are there and ship them as soon as they are there. Like this we also don't need a 10 minute default timeout but pick them up as soon as they are available.

Reply:
Thanks @ruflin for the review. As we discussed earlier, instead of waiting for 10 minutes to start collecting metrics, we should collect EC2 instance metrics but keep aws.ec2 metricset fields empty. I will make the changes in the next commit.

Reply:
I would prefer that way, probably users are interested in knowing about instances in states like `pending`, `stopping`, etc.

Reply:
Yep Nicolas brought up a good suggestion about adding a new metricset just for the detailed information about the instance itself and the status like `running`, `pending` or `stopping` will be a metric in this new metricset.

Reply:
why adding a new metricset for it? are the queries to gather this information different? new metricset means more queries == more spending. In general I would go that way only if we think periods should be different or collection method is different

Reply:
@exekias good point. More spending might stop users using this... Originally Nicolas and I are thinking collecting these metrics with a larger period(less collection) since the ec2 instance itself usually don't change very often. But since we are already making the api call here, maybe we should just add the data here instead of making a new metricset. @ruflin What you think?

Reply:
It all depends on the amount of data of course, but I would also add another point in favor of a single metricset: Users can use this data when querying the metrics (filtering and grouping by it)

Reply:
@kaiyan-sheng If I remember correctly, you mentioned that it is a different API call to get this data but that it might be free?

Reply:
@ruflin yes, describeInstances. This is a different API call than cloudwatch. But we are already calling it to list all the instances for each region. Sorry for the confusion here!!

Reply:
But that call we make only once on startup and not on each Fetch. So overall it would double the number of calls we make.

Reply:
@ruflin I think describeInstances we need to call for each fetch because there might be new instances added or old instances removed at any time. Am I misunderstanding something here?

Reply:
Do we call it at the moment for each fetch? As far as I understand the discussion here is it should be a separate metricset or not. If we have to make 2 API calls anyway to get the metrics then we can directly add it to the same metricset. If it's a second API call we only make for the state field, I would argue it should go into a separate metricset.

Reply:
Yes we are making 2 API calls already, so same metricset it is!